sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java

```diff
@@ -601,30 +604,26 @@ private static DefaultNetworkSpec convertNetwork(
                 // Note: Multiple VIPs may share prefixes with each other. For example if one wants the VIP hostnames,
                 // across multiple ports, to reflect the host that's serving the port.
 
-                NamedVIPSpec namedVIPSpec = new NamedVIPSpec(
-                        portValueBuilder.build(),
-                        role,
-                        preReservedRole,
-                        principal,
-                        rawPort.getEnvKey(),
-                        name,
-                        DcosConstants.DEFAULT_IP_PROTOCOL,
-                        visibility,
-                        StringUtils.isEmpty(rawVip.getPrefix()) ? name : rawVip.getPrefix(),
-                        rawVip.getPort(),
-                        networkNames);
-                portSpecs.add(namedVIPSpec);
+                portSpecBuilder = NamedVIPSpec.newBuilder()
+                        // NamedVIPSpec settings:
+                        .protocol(DcosConstants.DEFAULT_IP_PROTOCOL)
+                        .vipName(StringUtils.isEmpty(rawVip.getPrefix()) ? name : rawVip.getPrefix())
+                        .vipPort(rawVip.getPort());
             } else {
-                portSpecs.add(new PortSpec(
-                        portValueBuilder.build(),
-                        role,
-                        preReservedRole,
-                        principal,
-                        rawPort.getEnvKey(),
-                        name,
-                        visibility,
-                        networkNames));
+                portSpecBuilder = PortSpec.newBuilder();
```

Comment:
It may make it a little easier to read if we negate the conditional and add the shorter branch at the start.

Reply:
Good point, done